In [33]:
# from sklearn.datasets import make_multilabel_classification
from torch.utils.data import Dataset
import json
import numpy as np
import pandas as pd
import torch
import time
from tqdm import tqdm

In [34]:
# f = open("outputs/year_venue.json", 'r')
# year_venue = json.load(f)

# f = open("outputs/abstracts_title.json", 'r')
# abstracts_title = json.load(f)

# f = open("outputs/author.json", 'r')
# author = json.load(f)


f = open("outputs/year_venue_naive_bayes.json", 'r')
year_venue = json.load(f)

f = open("outputs/abstracts_title_all.json", 'r')
abstracts_title = json.load(f)

f = open("outputs/author_all.json", 'r')
author = json.load(f)

In [59]:
#################################
weight_sentence = 3  # abstracts - title
weight_author = 3  # authors
weight_year_venue = 0  # year - venue
sum_ = weight_author + weight_sentence + weight_year_venue
#################################

YEAR_VENUE_WEIGHT = weight_year_venue/sum_
COAUTHOR_WEIGHT = weight_author/sum_
SENTENCE_WEIGHT = weight_sentence/sum_

def get_weighted_value(i, j):
    val1 = year_venue[i][j] * YEAR_VENUE_WEIGHT
    val2 = author[i][j] * COAUTHOR_WEIGHT
    val3 = abstracts_title[i][j] * SENTENCE_WEIGHT
    
    return val1 + val2 + val3
    

In [64]:
THRESHOLD = 0.1

kaggle_predict = []

for i in range(800):
    
    pred = ""
    
    for j in range(100):
        
        val = get_weighted_value(str(i), str(j))
        
        if val >= THRESHOLD:
            pred += str(j) + " "
#             print(pred)
            
    if pred:
        kaggle_predict.append(pred[:-1])
    else:
        kaggle_predict.append("-1")

        

In [65]:
kaggle = pd.DataFrame(columns=["ID", "Predict"])
kaggle['Predict'] = kaggle_predict
kaggle['ID'] = kaggle.index
kaggle = kaggle.set_index('ID')
kaggle.to_csv("../../kaggle/predict_stacking2_10_4.csv")

In [66]:
count = 0
for i in kaggle['Predict']:
    if i != '-1':
        count += 1

count

573

In [67]:
kaggle

Predict
ID         
0        92
1         2
2        31
3      6 23
4        53
..      ...
795      54
796      97
797      13
798      71
799      53

[800 rows x 1 columns]

# Kaggle Predictions

- YEAR_VENUE_WEIGHT = 1/3
- COAUTHOR_WEIGHT = 1/3
- SENTENCE_WEIGHT = 1/3
    - f1 = 0.47230

# Kaggle Predictions - 2

- YEAR_VENUE_WEIGHT = 0.08884680485159918.
- COAUTHOR_WEIGHT = 0.5925861815691253
- SENTENCE_WEIGHT = 0.3185670135792754
    - f1 = 0.52080

In [49]:
print(YEAR_VENUE_WEIGHT)
print(COAUTHOR_WEIGHT)
print(SENTENCE_WEIGHT)

0.09421771031395267
0.43393907367832263
0.4718432160077247


# Kaggle Predictions - 3

- YEAR_VENUE_WEIGHT = 0.09421771031395267.
- COAUTHOR_WEIGHT = 0.43393907367832263
- SENTENCE_WEIGHT = 0.4718432160077247
- THRESHOLD = 0.3
    - f1 = 0.53130

In [43]:
a = [0.9566845690739418,
 0.9977403210423819,
 0.9990949067782368,
 0.9995765717884457,
 0.9996834779248579,
 0.9996834779248579,
 0.9997960250953476,
 0.9997906369281058,
 0.9997804862444465]

b = [0.9140499814085832,
 0.9172208109372708,
 0.9194803406690103,
 0.9028562145490067,
 0.9062611898144489,
 0.9100093884489224,
 0.9120535730073112,
 0.9109805733621669,
 0.9083259223343757]

c = [0.03217569098314048,
 0.033274246786204,
 0.09689634519674081,
 0.1266490732680644,
 0.18306285315357687,
 0.19963939094534452,
 0.1829821136259908,
 0.15826426166468657,
 0.1324441261793674]

In [46]:
max_ = 0
idx = 0
for i in range(len(a)):
    if a[i] + b[i] + c[i] >= max_:
        max_ = a[i] + b[i] + c[i]
        idx = i

In [48]:
thresholds = [0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6]
thresholds[5]

0.3

In [15]:
year_venue

{'0': {'0': 0.025694689151316938,
  '1': 1.0291680092039959,
  '2': 0.01986587913268543,
  '3': 0.029578283119578292,
  '4': 0.020024560958136185,
  '5': 0.01503550718119987,
  '6': 0.023882172471408825,
  '7': 0.004495304434844805,
  '8': 0.02106706354020333,
  '9': 0.0314561004120304,
  '10': 0.045420960885347646,
  '11': 0.012750465005301734,
  '12': 0.02223214463073519,
  '13': 0.033206226874371136,
  '14': 0.022786272667570735,
  '15': 0.0012276493926395395,
  '16': 0.02587387920194599,
  '17': 0.019410662178441458,
  '18': 0.020584219057140845,
  '19': 0.025026606294111763,
  '20': 0.04313005773221776,
  '21': 0.022692878680218406,
  '22': 0.015639734735175805,
  '23': 0.047384138725208066,
  '24': 0.01591729066898925,
  '25': 0.018888396620325577,
  '26': 0.023258184382815048,
  '27': 0.011984420032173402,
  '28': 0.015122692909050394,
  '29': 0.03781978307708881,
  '30': 0.03722388061895031,
  '31': 0.01619877165794127,
  '32': 0.01828793634436775,
  '33': 0.01026188022862961,
